src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -44,9 +53,52 @@ internal sealed class MethodBuilderImpl : MethodBuilder
         protected override GenericTypeParameterBuilder[] DefineGenericParametersCore(params string[] names) => throw new NotImplementedException();
         protected override ParameterBuilder DefineParameterCore(int position, ParameterAttributes attributes, string? strParamName) => throw new NotImplementedException();
         protected override ILGenerator GetILGeneratorCore(int size) => throw new NotImplementedException();
-        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute) => throw new NotImplementedException();
-        protected override void SetCustomAttributeCore(CustomAttributeBuilder customBuilder) => throw new NotImplementedException();
-        protected override void SetImplementationFlagsCore(MethodImplAttributes attributes) => throw new NotImplementedException();
+        protected override void SetCustomAttributeCore(ConstructorInfo con, ReadOnlySpan<byte> binaryAttribute)
+        {
+            if (!IsPseudoCustomAttribute(con.ReflectedType!.FullName!, con, binaryAttribute))
+            {
+                _customAttributes ??= new List<CustomAttributeWrapper>();
+                _customAttributes.Add(new CustomAttributeWrapper(con, binaryAttribute));
+            }
+        }
+
+        private bool IsPseudoCustomAttribute(string attributeName, ConstructorInfo con, ReadOnlySpan<byte> data)
```

Comment:
If you keep handling of the pseudo custom attributes in the separate method, the name should be same in all builders (it is `IsPseudoCustomAttribute` vs `IsPseudoAttribute` now) and the signature should be same too (it can be `(ConstructorInfo con, ReadOnlySpan<byte> data)` - the method can fetch the type name internally, no need to pass it from outside).

Reply:
I would use separate method, but could not find an appropriate/better name, so moving to same method with a comment.